### ISSUE TITLE
Resume doesn't work

### ISSUE THREAD

**Issue by mookiexl** [koda-dernet/Side-Step#46]
**Issue:** 
When resuming training, the loss jumps back up to ~1.5 and training starts again from scratch.
The UI shows the progress resuming (epoch, schedule) but the weights are reset.

**Expected Behavior:** 
Training should continue where it left off.

**Bug introduced in:**
Every version I tried has it.

**Notes:**
- Affects both XL and regular models
- Resume works correctly in the official ACE-step 1.5 Gradio app.

**System:**
Kubuntu Linux 25.10
Nvidia 4060Ti 16gb
Side-step installed on a NTFS drive

**Comment by mookiexl** [comment]
So it turns out the issue is mismatched key names. The fix is to add the following to trainer_helpers.py in resume_checkpoint (from line 745)

```
            PREFIX_TO_STRIP = "base_model.model."
            state_dict = {
                (k[len(PREFIX_TO_STRIP):] if k.startswith(PREFIX_TO_STRIP) else k): v
                for k, v in state_dict.items()
            }
            
            def _fix_lora_key(key):
                # If key doesn't already have .default suffix and has lora_A or lora_B, add it
                if ".default" not in key and ("lora_A" in key or "lora_B" in key):
                    # Insert .default before .weight
                    if key.endswith(".weight"):
                        return key[:-len(".weight")] + ".default.weight"
                    elif key.endswith(".bias"):
                        return key[:-len(".bias")] + ".default.bias"
                return key
            
            state_dict = {
                _fix_lora_key(k): v
                for k, v in state_dict.items()
            }
```

and from line 769 add:
```
            if hasattr(decoder, "base_model"):
                decoder = decoder.base_model
            if hasattr(decoder, "model"):
                decoder = decoder.model
```

After this the loss appears to be normal.
